sdcm/commit_log_check_thread.py

```diff
@@ -0,0 +1,192 @@
+import traceback
+import logging
+import time
+import re
+import threading
+from dataclasses import dataclass
+from distutils.util import strtobool
+from sdcm.sct_events.database import CommitLogCheckErrorEvent
+from sdcm.rest.remote_curl_client import RemoteCurlClient
+
+
+@dataclass
+class CommitlogConfigParams:
+    def __init__(self, db_cluster, ):
+        logger = logging.getLogger(self.__class__.__name__)
+        with db_cluster.cql_connection_patient(
+                node=db_cluster.nodes[0],
+                connect_timeout=300,) as session:
+            self.total_space = int(session.execute(
+                "SELECT value FROM system.config WHERE name='commitlog_total_space_in_mb'").one().value)
+            self.use_hard_size_limit = bool(strtobool(session.execute(
+                "SELECT value FROM system.config WHERE name='commitlog_use_hard_size_limit'").one().value))
+            self.segment_size_in_mb = int(session.execute(
+                "SELECT value FROM system.config WHERE name='commitlog_segment_size_in_mb'").one().value)
+            self.max_disk_size = int(RemoteCurlClient(
+                host="localhost:10000", endpoint="commitlog", node=db_cluster.nodes[0]).run_remoter_curl(
+                method="GET", path="metrics/max_disk_size", params=None).stdout)
+            self.smp = len(re.findall(
+                "shard",
+                db_cluster.nodes[0].remoter.run('seastar-cpu-map.sh -n scylla').stdout))
+            self.total_space = int(self.max_disk_size / self.smp)
+
+            logger.debug("CommitlogConfigParams")
+            logger.debug("smp: %s", self.smp)
+            logger.debug("max_disk_size: %s", self.max_disk_size)
+            logger.debug("total_space: %s", self.total_space)
+            logger.debug("use_hard_size_limit: %s", self.use_hard_size_limit)
+            logger.debug("segment_size_in_mb: %s", self.segment_size_in_mb)
+
+    smp: int
+    total_space: int
+    max_disk_size: int
+    use_hard_size_limit: bool
+    segment_size_in_mb: int
+
+
+@dataclass
+class PrometheusQueries:
+    def __init__(self, commitlog_params: CommitlogConfigParams):
+        self.overflow_commit_log_directory = self.overflow_commit_log_directory.format(
+            commitlog_total_space=commitlog_params.total_space,
+            commitlog_segment_size_in_mb=commitlog_params.segment_size_in_mb
+        )
+
+        self.zero_free_segments = self.zero_free_segments.format(
+            commitlog_total_space=commitlog_params.total_space,
+            commitlog_segment_size_in_mb=commitlog_params.segment_size_in_mb
+        )
+    # Prometheus queries with detailed description below
+    overflow_commit_log_directory = ("scylla_commitlog_disk_total_bytes>({commitlog_total_space}"
```

Comment:
If we want to be absolutely precise, we should have actually have a `>=` here (I might have missed it in the review of previous version)

Reply:
fixed